datashape/discovery.py

```diff
@@ -362,9 +362,10 @@ def unpack(ds):
         return ds
 
 
-@dispatch(dict)
-def discover(d):
-    return Record([[k, discover(d[k])] for k in sorted(d)])
+@discover.register(dict)
+@discover.register(MappingProxyType)
+def _mapping_discover(m):
+    return Record((k, discover(m[k])) for k in sorted(m))
```

Comment:
What if the user passes in an `ordereddict` and intentionally wants the dshape's fields' order to match the original?